### PR TITLE
[exo-deploy 3] Round off exo deploy

### PR DESCRIPTION
**Pull request by hugoesthere**
Run `terraform apply` and make final changes to Terraform files to make `exo deploy` work.

### CODE REVIEW

File: src/application/deployer.go [hunk 1]
```diff
@@ -53,5 +54,15 @@ func StartDeploy(deployConfig types.DeployConfig) error {
 	}
 
 	deployConfig.LogChannel <- "Planning deployment..."
-	return terraform.RunPlan(deployConfig, secrets)
+	terraform.RunPlan(deployConfig, secrets)
+	if err != nil {
+		return err
+	}
+
+	if ok := prompt.Confirm("Do you want to continue?"); ok {
```
Comment (charlierudolph) [inline src/application/deployer.go]: Can this prompt me more descriptive? This is confirming you want toi actually deploy correct? Also why is this prompt here versus at the beginning?

File: src/terraform/commands.go [hunk 2]
```diff
@@ -33,6 +33,21 @@ func RunPlan(deployConfig types.DeployConfig, secrets types.Secrets) error {
 	return err
 }
 
+// RunApply runs the 'terraform apply' command and passes variables in as command flags
+func RunApply(deployConfig types.DeployConfig, secrets types.Secrets) error {
+	command := []string{"terraform", "apply"}
```
Comment (charlierudolph) [inline src/terraform/commands.go]: Thoughts on removing this variable? Could do the following
```go
err = util.RunAndLog(deployConfig.TerraformDir, []string{}, deployConfig.LogChannel, "terraform", "apply", vars...)

File: src/terraform/commands.go [hunk 3]
```diff
@@ -33,6 +33,21 @@ func RunPlan(deployConfig types.DeployConfig, secrets types.Secrets) error {
...
+	vars, err := CompileVarFlags(deployConfig, secrets)
+	if err != nil {
+		return err
+	}
+	command = append(command, vars...)
+	err = util.RunAndLog(deployConfig.TerraformDir, []string{}, deployConfig.LogChannel, command...)
+	if err != nil {
+		return errors.Wrap(err, "'terraform apply' failed")
```
Comment (charlierudolph) [inline src/terraform/commands.go]: I don't think this wrap is needed as its wrapping it with arguments passed into `util.RunAndLog`. Ideally we only wrap it with context not passed to the function and rely on the function to report good errors based on its inputs. 

`util.RunAndLog` already wraps run errors with the command it was running and the output

File: src/application/deployer.go (hunk 1 above)
Comment (hugoesthere) [reply src/application/deployer.go]: Yes it's prompting to deploy, it's asked after `terraform plan` and before `terraform apply`

**Review by charlierudolph — APPROVED** [review]

**Pull request closed without merging**